File: src/types/portfolioProject.ts
export interface PortfolioProject {
  id: string;
  name: string;
  category: string;
  description: string;
  techStack: string[];
  date: string;
  dateValue: string; // ISO-sortable value (e.g. "2024-05"), used for sorting by date
  duration: string;
  url: string;
  bannerClassName: string; // Tailwind background classes for the card banner
}

File: backend/app/schemas/portfolio_project.py
from pydantic import BaseModel


class PortfolioProject(BaseModel):
    id: str
    name: str
    category: str
    description: str
    techStack: list[str]
    date: str
    dateValue: str
    duration: str
    url: str
    bannerClassName: str
    imageUrl: str | None = None
